mala/datahandling/snapshot.py

```diff
@@ -73,7 +76,65 @@ def __init__(self, input_npy_file, input_npy_directory,
 
         # Legacy functionality: Determine whether the snapshot contains
         # numpy or HDF5 files.
-        self.snapshot_type = snapshot_type
+        self.snapshot_type = snapshot_type        
+
+        # All the dimensionalities of the snapshot.
+        self.grid_dimensions = None
+        self.grid_size = None
+        self.input_dimension = None
+        self.output_dimension = None
+        
+    def load_dimensions(self, descriptors_contain_xyz,
```

Comment:
This function should be obsolete now. The PhysicalData class (which all targets and descriptors inherit from) have `load_dimensions` functions for both OpenPMD and numpy and I would like to keep using those since we then only have to maintain code at one location. 

This probably requires some changes in the DataHandler again, as the snapshot locations have to be passed to the calculators. Sorry for the mess, I added this function (I think) before the PhysicalData class was introduced.